packages/shared/src/components/auth/common.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { CSSProperties, forwardRef, useContext } from 'react';
+import classNames from 'classnames';
+import TwitterIcon from '../icons/Twitter';
+import FacebookIcon from '../icons/Facebook';
+import GoogleIcon from '../icons/Google';
+import GitHubIcon from '../icons/GitHub';
+import AppleIcon from '../icons/Apple';
+import classed from '../../lib/classed';
+import { IconType } from '../buttons/Button';
+import FeaturesContext from '../../contexts/FeaturesContext';
+
+export interface Provider {
+  icon: IconType;
+  provider: string;
+  onClick?: () => unknown;
+  className?: string;
+  style?: CSSProperties;
+}
+
+export const providerMap: Record<string, Provider> = {
+  twitter: {
+    icon: <TwitterIcon />,
+    provider: 'Twitter',
+    style: { backgroundColor: '#1D9BF0' },
+  },
+  facebook: {
+    icon: <FacebookIcon />,
+    provider: 'Facebook',
+    style: { backgroundColor: '#4363B6' },
+  },
+  google: {
+    icon: <GoogleIcon secondary />,
+    provider: 'Google',
+    style: { backgroundColor: '#FFFFFF', color: '#0E1217' },
+  },
+  gitHub: {
+    icon: <GitHubIcon />,
+    provider: 'GitHub',
+    style: { backgroundColor: '#2D313A' },
+  },
+  apple: {
+    icon: <AppleIcon />,
+    provider: 'Apple',
+    style: { backgroundColor: '#0E1217' },
+  },
+};
```

Comment:
Utilize existing color classes introduced in a recently merged PR for social providers.